File: April Challenge 2021 Division 3/WorthyMatrix.py
# My solution for worthy matrix problem


import numpy as np
for _ in range(int(input())):
    N,M,K = list(map(int,input().split()))
    mat = np.zeros((N+1,M+1))
    for i in range(1,N+1):
        temp = list(map(int,input().split()))
        for j in range(1,M+1):
            mat[i][j] = temp[j-1]

    result = 0   
    for i in range(N+1):
        cumSum = 0
        for j in range(M+1):
            mat[i][j] += cumSum
            cumSum = mat[i][j]
            
    for j in range(M+1):
        cumSum = 0
        for i in range(N+1):
            mat[i][j] += cumSum
            cumSum = mat[i][j]
            
    Z = min(N,M)
    for l in range(1,Z+1):
        for i in range(l,N+1):
            for j in range(l,M+1):
                if ((mat[i,j]+mat[i-l,j-l]-mat[i-l,j]-mat[i,j-l])/(l*l)) >= K:
                    result += 1
    # for i2 in range(N):
    #     for j2 in range(M):
    #         for i1 in range(i2+1):
    #             for j1 in range(j2+1):
    #                 if i1-i2 == j1-j2:
    #                     res = mat[i2,j2]
    #                     if i1>0:
    #                         res = res - mat[i1-1,j2]
    #                     if j1 >0:
    #                         res = res - mat[i2,j1-1]
    #                     if i1 > 0 and j1 > 0:
    #                         res += mat[i1-1,j1-1]
    #                     res = res/((i2-i1+1)*(j2-j1+1))
    #                     if res >= K:
    #                         result += 1
                            
    print(result)
    
 

    
    
        
    

File: April Challenge 2021 Division 3/WorthyMatrix.cpp
// https://www.codechef.com/APRIL21C/problems/KAVGMAT


#include <bits/stdc++.h>
#include <iostream>
using namespace std;
typedef long long int ll;

ll min(ll a, ll b)
{
    if(a<b)
    {
        return a;
    }
    else
    {
        return b;
    }
}

int main() {
	ll t;
	cin >> t;
	while(t--)
	{
	    ll N,M,K;
	    cin >> N >> M >> K;
	    double mat[N+1][M+1];
	    for(ll i=0;i<=N;i++)
	    {
	        for(ll j = 0;j<=M;j++)
	        {
	            if(i == 0 || j == 0)
	            {
	                mat[i][j] = 0;
	            }
	            else
	            {
	                cin >> mat[i][j]; 
	            }
	        }
	    }
	    
	    for(ll i=0;i<=N;i++)
	    {
	        ll cumSum = 0;
	        for(ll j = 0;j<=M;j++)
	        {
	            cumSum += mat[i][j];
	            mat[i][j] = cumSum;
	        }
	    }	    

	    for(ll j=0;j<=M;j++)
	    {
	        ll cumSum = 0;
	        for(ll i = 0;i<=N;i++)
	        {
	            cumSum += mat[i][j];
	            mat[i][j] = cumSum;
	            //cout << mat[i][j] ;
	        }
	        //cout << endl;
	    }
	    
	    ll result = 0;
	    ll Z = min(N,M);
	    for(ll l=1;l<=Z;l++)
	    {
	        for(ll i=l;i<=N;i++)
	        {
	            for(ll j=l;j<=M;j++)
	            {
	                if(((mat[i][j]+mat[i-l][j-l]-mat[i-l][j]-mat[i][j-l])/(l*l)) >= K)
	                {
	                    result++;
	                }
	            }
	        }
	    }
	    cout << result << endl;
	    
	    
	}
	return 0;
}
